### PR TITLE
Feature/task 5660 qr code list

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Shows a list of QR codes and a list of checkins in their respective places. This contains some dummy data that will be removed once the creation of TraceLocations is implemented.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5660
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5657

## Screenshots
QR Codes
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 35 02
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 35 32
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 37 59
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 38 10
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 35 53
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 36 02
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 36 17
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 36 28

Check-ins
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 36 42
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 39 23
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 39 34
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 39 43
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 37 09
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 37 17
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 40 03
Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-18 at 17 40 13

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/App/AppNavigationController.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import Foundation
 import UIKit
 
-class AppNavigationController: ENANavigationControllerWithFooter {
+class AppNavigationController: UINavigationController {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/App/AppNavigationController.swift:8]: 👏

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/ConfigureableCell.swift [hunk 2]
```diff
@@ -4,14 +4,19 @@
 
 import Foundation
 
-protocol ConfigureableCell {
+protocol ConfigureableCell: ReuseIdentifierProviding {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/ConfigureableCell.swift:7]: Protocol Inception, not sure if I really like this.

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/Overview/CheckinsOverviewViewController.swift [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+			onButtonTap: { [weak self] in
+				self?.viewModel.didTapEntryCellButton(at: indexPath)
+			}
+		)
+
+		return cell
+	}
+
+	private func updateEmptyState() {
+		tableView.backgroundView = viewModel.isEmpty ? EmptyStateView(viewModel: CheckinsOverviewEmptyStateViewModel()) : nil
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/Overview/CheckinsOverviewViewController.swift:261]: I know its convenient to use the backgroundView but it also leads to stuff like this:
Simulator Screen Shot - iPhone 12 mini - 2021-03-18 at 18 10 22 Would look even worse on smaller screens.

**Review by marcussc — COMMENTED** [review]
What about UI-Tests/Screenshots? :)

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 4]
```diff
@@ -1928,13 +1938,96 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
...
+
+"TraceLocations_Overview_Title" = "Meine QR-Codes";
+
+"TraceLocations_Overview_MenuButtonTitle" = "Menü";
+
+"TraceLocations_Overview_AddButtonTitle" = "QR-Code erstellen";
+
+"TraceLocations_Overview_EmptyTitle" = "Noch keine QR-Codes erstellt";
+
+"TraceLocations_Overview_EmptyDescription" = "Hier werden alle QR-Codes angezeigt, die Sie für ein Ort oder Event erstellt haben. Sie können die QR-Codes löschen, wenn diese nicht mehr verwendet werden sollen.";
```
Comment (tcfos) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2003]:
```suggestion
"TraceLocations_Overview_EmptyDescription" = "Hier werden alle QR-Codes angezeigt, die Sie für einen Ort oder ein Event erstellt haben. Sie können die QR-Codes löschen, wenn diese nicht mehr verwendet werden sollen.";
```

**Review by janetback — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 5]
```diff
@@ -1928,13 +1938,96 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
...
+
+"Checkins_Overview_DeleteOne_AlertMessage" = "Sie können dann Personen, die zur selben Zeit für diesen Ort oder Event eingecheckt waren, bei Bedarf nicht mehr warnen oder von ihnen gewarnt werden.";
+
+"Checkins_Overview_DeleteOne_AlertConfirmButtonTitle" = "Entfernen";
+
+"Checkins_Overview_DeleteOne_AlertCancelButtonTitle" = "Abbrechen";
+
+"Checkins_Overview_DeleteAll_AlertTitle" = "Wollen Sie wirklich alle Einträge entfernen?";
+
+"Checkins_Overview_DeleteAll_AlertMessage" = "Sie können dann Personen, die zur selben Zeit für diesen Ort oder Event eingecheckt waren, bei Bedarf nicht mehr warnen oder von ihnen gewarnt werden.";
```
Comment (tcfos) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:1981]:
```suggestion
"Checkins_Overview_DeleteAll_AlertMessage" = "Sie können dann Personen, die zur selben Zeit für diese Orte oder Events eingecheckt waren, bei Bedarf nicht mehr warnen oder von ihnen gewarnt werden.";
```

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 6]
```diff
@@ -1928,13 +1938,96 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
...
+
+"TraceLocations_Overview_DeleteOne_AlertMessage" = "Bitte denken Sie daran, dass der QR-Code danach nicht mehr zum Einchecken verwendet werden kann.";
+
+"TraceLocations_Overview_DeleteOne_AlertConfirmButtonTitle" = "Löschen";
+
+"TraceLocations_Overview_DeleteOne_AlertCancelButtonTitle" = "Abbrechen";
+
+"TraceLocations_Overview_DeleteAll_AlertTitle" = "Wollen Sie wirklich alle QR-Codes löschen?";
+
+"TraceLocations_Overview_DeleteAll_AlertMessage" = "Bitte denken Sie daran, dass der QR-Code danach nicht mehr zum Einchecken verwendet werden kann.";
```
Comment (tcfos) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2021]:
```suggestion
"TraceLocations_Overview_DeleteAll_AlertMessage" = "Bitte denken Sie daran, dass diese QR-Codes danach nicht mehr zum Einchecken verwendet werden können.";
```

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 4 above)
Comment (janetback) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: Hi @tcfos - I have corrected this.

**Review by tcfos — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Overview/TraceLocationsOverviewViewController.swift [hunk 7]
```diff
@@ -234,7 +234,17 @@ class TraceLocationsOverviewViewController: UITableViewController, FooterViewHan
 	}
 
 	private func updateEmptyState() {
-		tableView.backgroundView = viewModel.isEmpty ? EmptyStateView(viewModel: TraceLocationsOverviewEmptyStateViewModel()) : nil
+		let emptyStateView = EmptyStateView(viewModel: TraceLocationsOverviewEmptyStateViewModel())
+
+		emptyStateView.additionalTopPadding = tableView.rectForRow(at: IndexPath(row: 0, section: 0)).maxY
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Overview/TraceLocationsOverviewViewController.swift:239]: Can you add some comments on why and what is happnening here?

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Overview/TraceLocationsOverviewViewController.swift (hunk 7 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Overview/TraceLocationsOverviewViewController.swift:239]: Done

**Pull request merged**